lib/taskana-core/src/main/java/pro/taskana/configuration/TaskanaEngineConfiguration.java

```diff
@@ -298,36 +299,27 @@ private void initSchemaName(String schemaName) {
     }
 
     private void initTaskanaRoles(Properties props, String rolesSeparator) {
-        List<String> validPropertyNames = Arrays.stream(TaskanaRole.values())
-            .map(TaskanaRole::getPropertyName)
-            .collect(Collectors.toList());
-        for (Object obj : props.keySet()) {
-            String propertyName = ((String) obj);
-            if (validPropertyNames.contains(propertyName.toLowerCase().trim())) {
-                String propertyValue = props.getProperty(propertyName);
-                Set<String> roleMemberSet = new HashSet<>();
-                StringTokenizer st = new StringTokenizer(propertyValue, rolesSeparator);
-                while (st.hasMoreTokens()) {
-                    String token = st.nextToken().toLowerCase().trim();
-                    roleMemberSet.add(token);
-                }
-                TaskanaRole key = TaskanaRole.fromPropertyName(propertyName);
-                if (key != null) {
-                    roleMap.put(key, roleMemberSet);
-                } else {
-                    LOGGER.error("Internal System error when processing role property {}.", propertyName);
-                    throw new SystemException(
-                        "Internal System error when processing role property " + propertyName);
-                }
-            }
-        }
+        List<String> validPropertyNames = TaskanaRole.getValidPropertyNames();
+
+        props.keySet()
+            .stream()
+            .map(key -> (String) key)
+            .filter(propertyName -> validPropertyNames.contains(propertyName.toLowerCase().trim()))
+            .forEach(validPropertyName -> roleMap.put(TaskanaRole.fromPropertyName(validPropertyName),
+                getTokensWithCollection(props.getProperty(validPropertyName), rolesSeparator)));
+
         ensureRoleMapIsFullyInitialized();
 
         if (LOGGER.isDebugEnabled()) {
             roleMap.forEach(
                 (k, v) -> LOGGER.debug("Found Taskana RoleConfig {} : {} ", k, LoggerUtils.setToString(v)));
         }
+    }
 
+    private HashSet<String> getTokensWithCollection(String str, String rolesSeparator) {
+        return Collections.list(new StringTokenizer(str, rolesSeparator)).stream()
+            .map(token -> ((String) token).toLowerCase().trim())
```

Comment:
Same as above. I do not like casts. String::valueof

Reply:
;)

Reply:
Cast is still there :D